src/Repository/DefaultUnleashProxyRepository.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Unleash\Client\Repository;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Symfony\Component\VarExporter\Exception\LogicException;
+use Unleash\Client\Configuration\Context;
+use Unleash\Client\Configuration\UnleashConfiguration;
+use Unleash\Client\Configuration\UnleashContext;
+use Unleash\Client\DTO\DefaultProxyFeature;
+use Unleash\Client\DTO\Feature;
+use Unleash\Client\DTO\ProxyFeature;
+
+final class DefaultUnleashProxyRepository implements ProxyRepository
+{
+    public function __construct(
+        private UnleashConfiguration $configuration,
+        private ClientInterface $httpClient,
+        private RequestFactoryInterface $requestFactory
+    ) {
+    }
+
+    public function findFeatureByContext(string $featureName, ?Context $context = null): ?ProxyFeature
+    {
+        $cacheKey = $this->getCacheKey($featureName, $context);
+
+        if ($this->configuration->getCache() !== null && $this->configuration->getCache()->has($cacheKey)) {
+            $cachedFeature = $this->configuration->getCache()->get($cacheKey);
+            if (is_array($cachedFeature)) {
+                $featureData = $this->validateResponse($cachedFeature);
+                if ($featureData !== null) {
+                    return new DefaultProxyFeature($featureData);
+                }
+            }
+        }
+
+        $context ??= new UnleashContext();
+        $featureUrl = $this->configuration->getUrl() . 'frontend/features/' . $featureName;
+        $url = $this->addQuery($featureUrl, $this->contextToQueryString($context));
+
+        $request = $this->requestFactory->createRequest('GET', $url)
+            ->withHeader('Content-Type', 'application/json')
+            ->withHeader('Accept', 'application/json');
+
+        foreach ($this->configuration->getHeaders() as $name => $value) {
+            $request = $request->withHeader($name, $value);
+        }
+
+        $apiKey = $this->configuration->getProxyKey();
+        if ($apiKey === null) {
+            // The only way we can get here is if this is manually constructed without the builder
+            // @codeCoverageIgnoreStart
+            throw new LogicException('No api proxy key was specified');
+            // @codeCoverageIgnoreEnd
+        }
```

Comment:
Details: maybe we should check this a bit earlier in the method, but it's an edge case, so not a big deal if you just ignore my comment

Reply:
Easy enough to change, good idea, I'll patch this